CMakeLists.txt

```diff
@@ -138,6 +138,7 @@ sls_enable_cxx_warning("-Wduplicated-cond")
 sls_disable_cxx_warning("-Wclass-memaccess")
 sls_disable_c_warning("-Wstringop-truncation")
 sls_disable_c_warning("-Wformat-nonliteral")
```

Comment:
It is also quite easy to enable this warning. I didn't find anywhere where we reuse the char* to format. It's not 100% related to this PR but having this warning on might be nice.

Reply:
I added this before 6.0.0 was released. It was to do with the snprintf combining mess and retvals that were both the same size, even though we were dealing with it

Reply:
removed both warnings and fixed it in places.